mysql-test/suite/binlog/t/percona_bug1733781.test

```diff
@@ -0,0 +1,17 @@
+--source include/have_log_bin.inc
+--source include/have_innodb.inc
+
+--echo #
+--echo # Bug 1733781: sql/handler.cc:3008: int handler::ha_close(): Assertion `m_lock_type == 2' failed
+--echo #
+
+--let $assert_text= server-id must be zero
+--let $assert_cond= @@server_id = 0
+--source include/assert.inc
+
+CREATE TEMPORARY TABLE t4(f int) ENGINE=INNODB;
+LOCK TABLES t4 WRITE;
+# Format description event that forces closing of open temp tables
+BINLOG ' SOgWTg8BAAAAbgAAAHIAAAAAAAQANS42LjMtbTUtZGVidWctbG9nAAAAAAAAAAAAAAAAAAAAAAAA AAAAAAAAAAAAAAAAAABI6BZOEzgNAAgAEgAEBAQEEgAAVgAEGggAAAAICAgCAAAAAAVAYI8=';
```

Comment:
In one of the tests I did (`main.bug75721`), changing `BINLOG ' SOgWTg8B` to `BINLOG ' SOgWTg8C` (changing 8th character) was enough to make `server-id=1`. So you can potentially get rid of `.opt` file and add this test to 5.6 as well.

Reply:
This is great - it crashed 5.6 (but not 5.5) too!